cli/kaos_cli/services/tests/test_state_service.py

```diff
@@ -0,0 +1,228 @@
+import os
+import time
+from parameterized import parameterized
+from configobj import ConfigObj
+
+from unittest import TestCase
+
+from kaos_cli.services.state_service import StateService
+from kaos_cli.constants import CONFIG_PATH, KAOS_STATE_DIR
+
+
+def state_service_test_inputs():
+    return [
+        (
+            'context1', 'param1'
+        ),
+        (
+            'context2', 'param2'
+        ),
+        (
+            '', 'param3'
+        ),
+        (
+            'context4', ''
+        ),
+        (
+            '', ''
+        ),
+        (
+            {}, 'param6'
+        ),
+        (
+            'context7', {}
+        ),
+        (
+            {}, {}
+        ),
+        (
+            '', {}
+        ),
+        (
+            {}, ''
+        ),
+        (
+            None, 'param11'
+        ),
+        (
+            'context11', None
+        ),
+        (
+            None, None
+        ),
+        (
+            {}, None
+        ),
+        (
+            None, {}
+        ),
+        (
+            '', None
+        ),
+        (
+            None, ''
+        ),
+        (
+            1, 'param18'
+        ),
+        (
+            'context', 2
+        ),
+        (
+            3, {}
+        ),
+        (
+            {}, 4
+        ),
+        (
+            5, ''
+        ),
+        (
+            '', 6
+        ),
+        (
+            7, None
+        ),
+        (
+            None, 8
+        ),
+
+    ]
+
+
+class TestStateService(TestCase):
+
+    # Arrange
+    TestCase.state_service = StateService()
+    TestCase.switch = False
+
+    def test_state_service_create(self):
+
+        # Act
+        TestCase.state_service.create()
+
+        # Assert
+        self.assertTrue(os.path.exists(KAOS_STATE_DIR))
+
+    def test_state_service_is_created(self):
+
+        # Act
+        TestCase.state_service.is_created(KAOS_STATE_DIR)
+
+        # Assert
+        self.assertEqual(True, TestCase.state_service.is_created(KAOS_STATE_DIR))
+
+    def test_state_service_list_providers(self):
+
+        # Act
+        provider_list = TestCase.state_service.list_providers()
+
+        # Assert
+        self.assertEqual([], provider_list)
+
+    @parameterized.expand(state_service_test_inputs)
+    def test_state_service_set_and_get(self, context, param):
+
+        # Arrange
+        if not isinstance(context, str):
+            TestCase.switch = True
+            if context is None:
+                context = {}
+            else:
+                context = str(context)
```

Comment:
This is an anti-pattern. Please create an instance of the `StateService` in each unit test and carry out the mutations you're testing for inside that scope of the test method only.

Reply:
Has been done as suggested.